basic_autonomy/src/basic_autonomy.cpp

```diff
@@ -870,6 +870,14 @@ namespace basic_autonomy
             lanelet::BasicLineString2d new_points = create_lanechange_path(lanelets_in_path[lane_change_iteration],lanelets_in_path[lane_change_iteration + 1]);
             centerline_points.insert(centerline_points.end(), new_points.begin(), new_points.end());
 
+            //Add points from following lanelet to provide sufficient distance for adding buffer
+            auto following_lanelets = wm->getMapRoutingGraph()->following(lanelets_in_path[lane_change_iteration + 1], false);
```

Comment:
Does lanelets_in_path[lane_change_iteration + 1 give us the straight lanelet following the lanechange?

Reply:
lanelets_in_path[lane_change_iteration + 1, gives the last lanelet involved in lane change. We are using the lanelet following() that which is the straight lanelet following the lanelet